services/QuillLMS/client/app/bundles/Teacher/components/assignment_flow/unit_templates_manager/unit_templates_manager.jsx

```diff
@@ -106,20 +92,25 @@ export default class UnitTemplatesManager extends React.Component {
   filterModels = (category, grade, typeId, gradeLevelRange) => {
     const { unitTemplatesManager, } = this.state
     let displayedModels = unitTemplatesManager.models
-    let selectedCategoryId
     if (grade) {
       displayedModels = this.modelsInGrade(grade)
     }
     if (category) {
-      const categoryName = category.toUpperCase() === 'ELL' ? category.toUpperCase() : category
-      selectedCategoryId = unitTemplatesManager.categories.find(cat => cat.name === categoryName).id
-      displayedModels = displayedModels.filter(ut =>
-        ut.unit_template_category.name === categoryName
-      )
+      displayedModels = displayedModels.filter(ut => ut.unit_template_category.name === category)
     }
     if (typeId) {
-      const selectedTypeName = types.find(t => t.id === typeId).name
-      displayedModels = displayedModels.filter(ut => ut.type.name === selectedTypeName)
+      const selectedType = ACTIVITY_PACK_TYPES.find(t => t.id === typeId)
+      const { name } = selectedType
+      displayedModels = displayedModels.filter(ut => {
+        const { type, unit_template_category } = ut
+        if(selectedType.types || typeId === 'independent-practice') {
```

Comment:
Does it make sense to also check the `|| typeId === 'independent-practice'` bit?  That type should always have `types`, right?  And even if it was somehow missing, that would make this `if` block error since if `types` is `null`, the `includes` call will crash?

Reply:
good catch, this should just be `typeId === 'independent-practice'`

Reply:
An alternative thing to do here, if you feel like it makes sense, is to just make this `selectedType.types` as that would make this code automatically trigger if we ever added `types` to a type other than `indepdendent-practice`.  Could go either way.